src/System.Text.Json/tests/Utf8JsonReaderTests.MultiSegment.cs

```diff
@@ -1729,6 +1729,10 @@ private static void TrailingCommasHelper(ReadOnlySequence<byte> utf8, JsonReader
         }
 
         [Theory]
+        [InlineData("{]")]
+        [InlineData("[}")]
+        [InlineData("{// comment\n]")]
+        [InlineData("[// comment\n}")]
```

Comment:
Note: These new tests don't fail without the fix, but just adding them for completeness.

The `Debug.Asserts` though might be unclear, were still technically accurate.

It could say both/either since both `first == JsonConstants.CloseBrace` and `first == JsonConstants.CloseBracket` conditions are checked previously (which is why this assert never failed with these new or existing tests).